Cloudwatch: Upgrade aws-sdk and display external ids for temporary credentials

sarahzinger opened the pull request:
**What is this feature?**

We are working on adding support for temporary credentials in Grafana Cloud for the Cloudwatch Datasource. As part of that work we'll need to display an externalId to users so that they can create an IAM role in AWS. This update will display that id to users when the feature toggle is on.

This pr contains the following changes to make that possible:
- I noticed that I wasn't able to pick up an externalId from a custom.ini file which was my intention in merging https://github.com/grafana/grafana/pull/72062 I'm not sure how I didn't notice it before, but it seems it is necessary afterall to explicitly set the property in the environment? Or maybe there was some other step in there that I missed? I'm trying to follow the existing patterns I see elsewhere.
- Use the latest version of the grafana-aws-sdk-react package which contains the frontend changes necessary: https://github.com/grafana/grafana-aws-sdk-react/pull/54
- ~There have been [a few other changes](https://github.com/grafana/grafana-aws-sdk-react/compare/9e29ac68c8e2c7d51a5bf71676ac838744800d55...main) in that package as well since it was last installed in grafana. One of those changes seems to be causing one of our frontend tests to fail unless I manually enable something in our config. This test is to support a deprecated auth type that has been deprecated since 7.3. I don't believe we're causing breaking changes here, but anytime we have to change code to make a test pass, it does make me worry a bit. If users do run into this I believe all they'll need to do is enable a flag in their custom.ini, as we have done in this test. I do not think there are any cloud customers who could be affected.~
- Adds backend code to the cloudwatch plugin create an endpoint for our plugin's frontend to ping to fetch the external id. In the future it might be worth exploring doing this in the grafana-aws-sdk although I think this is a reasonable approach for now. 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

sarahzinger committed:
WIP: display external id

sarahzinger committed:
Upgrade package

sarahzinger committed:
Add missing env variable setting

sarahzinger committed:
Fix test

fridgepoet commented on pkg/tsdb/cloudwatch/routes/externalId.go:
I'd suggest external_id.go for the file name.
Also what do you think about adding a simple test for this one?

idastambuk commented:
I know why we had to change a test. I committed a line in a refactor that was actually a change in logic: https://github.com/grafana/grafana-aws-sdk-react/pull/50/files#diff-9bd5c403af851469a03ce8274d95b0f68667afc08c9b8c4a595e1d11092ba111L45
In the Cloudwatch test, since we didn't use to define the awsAssumeRoleEnabled field, it defaulted to true. But now after the refactor it defaults to false. 
Created a pull request [here](https://github.com/grafana/grafana-aws-sdk-react/pull/57) to fix it. Sorry about that!

sarahzinger committed:
Fixes after CR:
- change casing of route and file name
- add a test
- remove test changes in frontend, will instead install latest version of aws-sdk after release

sarahzinger committed:
Upgrade @grafana/aws-sdk to 0.1.1

sarahzinger committed:
Add back test changes

sarahzinger committed:
Set as true, since that's what the default in in defaults.ini

sarahzinger commented:
@idastambuk so after installing the latest it still breaks the test, I think that the issue is that we used to be pulling from boot settings, and now we're pulling from config and for some reason in the test env the default is that assumeRole is explicitly set to off. However IRL, it seems like the default.ini has set it to true, so I don't think this is really an issue users would run into. I have modified the test to explicitly setAssumeRole to true since that's what's in defaults.ini.

sarahzinger committed:
merge issue with betterer
